utils/funcexp/func_reverse.cpp

```diff
@@ -73,13 +74,14 @@ std::string Func_reverse::getStrVal(rowgroup::Row& row,
     // string assignment and implement a ref-count. Reversing in the
     // string buffer has the affect of reversing all strings from
     // which this one derived.
-    int len = str.length();
-    char* pbuf = new char[len + 1];
-    strncpy(pbuf, str.c_str(), len);
-    pbuf[len] = 0;
-    char* end = pbuf + len - 1;
-    reverse(pbuf, end);
-    string rstr = pbuf;
+    wstring wstr = utf8::utf8_to_wstring(str);
+    int wlen = wstr.length();
+    wchar_t* pbuf = new wchar_t[wlen + 1];
+    pbuf[wlen] = 0;
+    wchar_t* end = pbuf + wlen - 1;
+    reverse(wstr.c_str(), wstr.c_str() + wlen - 1, pbuf, end);
+    wstring wrstr = pbuf;
```

Comment:
I wonder if we really need this assignment  or we could use wrstr to instantiate rstr?

Reply:
Do you mean get rid of pbuf and write the reversed string directly into wrstr?